spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceBucket.java

```diff
@@ -110,11 +125,16 @@ public long lastModified() {
 	}
 
 	@Override
-	public Resource createRelative(String relativePath) throws IOException {
-		return new GoogleStorageResourceObject(this.storage,
+	public GoogleStorageResourceObject createRelative(String relativePath) throws IOException {
+		GoogleStorageResourceObject resource = new GoogleStorageResourceObject(this.storage,
 				getURI().toString() + (relativePath.startsWith("/") ? "" : "/")
 						+ relativePath,
-				this.createBucketIfRequired);
+				this.createBucketIfNotExists);
```

Comment:
This doesn't quiet make sense on the surface. You might want to let the user pass `createBlobIfNotExists` to `GoogleStorageResourceBucket`, or at least clearly document that auto-creation setting is inherited by the `GoogleStorageResourceObject`.

Reply:
That's not in the scope of this PR.
Please file a separate issue to fix that, if you feel it's necessary.

Reply:
Can we at least document the current behavior in the javadoc?

Reply:
Done.